### PR TITLE
fix: fix wrong return types + add missing fields

### PR DESCRIPTION
**Pull request by smoya**
**Description**

https://github.com/asyncapi/event-gateway is already implementing partially this API. During the development, I found out some return types are wrong and a missing method on Schema.

### CODE REVIEW

**Review by fmvilas — APPROVED** [review]

**Comment by fmvilas** [comment]
Good catch 🦅 👀

**Review by magicmatatjahu — CHANGES_REQUESTED** [review]

File: docs/v1.md [hunk 1]
```diff
@@ -208,7 +208,7 @@
 
 ## Schema
 - $id() : `string`
-- additionalItems : `Schema`
+- additionalItems : `boolean | Schema`
 - additionalProperties : `boolean | Schema`
 - allOf : `Schema`
 - anyOf : `Schema`
```
[the remark this answers is not in the document]
Comment (smoya) [reply docs/v1.md]: Good catch, fixed!

File: docs/v1.md [hunk 2]
```diff
@@ -219,6 +219,7 @@
 - contentMediaType : `string`
 - default : `any`
 - definitions : `Map[string, Schema]`
+- description : `string` | `undefined`
```
[the remark this answers is not in the document]
Comment (smoya) [reply docs/v1.md:222]: I ended up adding the `| undefined` to all of those for consistency

File: docs/v1.md [hunk 3]
```diff
@@ -207,60 +207,61 @@
...
+- $id() : `string` | `undefined`
+- additionalItems : `boolean | Schema`
 - additionalProperties : `boolean | Schema`
-- allOf : `Schema`
-- anyOf : `Schema`
+- allOf : `Schema[]`
+- anyOf : `Schema[]`
 - circularProps : `string[]`
-- const : `any`
+- const : `string` | `undefined`
```
Comment (smoya) [inline docs/v1.md:216]: I noticed this is not `any` but string as I saw in the JSON Schema spec.

Comment (magicmatatjahu) [reply docs/v1.md:216]: For `const` you can also define number value or boolean, so I would leave `any`

File: docs/v1.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply docs/v1.md:222]: Be aware that `undefined` is only in JS, so in (eg) GoLang you will end with `nil`. @jonaslagoni You defined also this API, do you accept changes?

[the remark this answers is not in the document]
Comment (jonaslagoni) [reply docs/v1.md:222]: Yea we should add `| undefined` to all Schema keywords, as none are required 🙂

[the remark this answers is not in the document]
Comment (smoya) [reply docs/v1.md:222]: `undefined` is the word we decided to use as `null`, `nil`, `none`, etc.

[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply docs/v1.md:222]: So if you wanna doc that everything is optional, maybe little `NOTE:` with information that all functions can also return null, where something isn't defined, will be better? Then you don't need to define this union with `null`.

[the remark this answers is not in the document]
Comment (smoya) [reply docs/v1.md:222]: I rather prefer being explicit on the signature of each method and be clear on the return types. Confusion can't happen then.

[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply docs/v1.md:222]: Ok :) But I have some problem here. In some places in the ParserJS we return in the array case the empty array (in object case, the empty object) if something isn't defined - for example, `properties()` always returns at least the empty object -> https://github.com/asyncapi/parser-js/blob/master/lib/models/schema.js#L192 so updated API won't be compatible with ParserJS

[the remark this answers is not in the document]
Comment (smoya) [reply docs/v1.md:222]: Without moving into details of why comparing to `null` is not the same as empty objects (no matters the language), we built this API definition for simplicity and looking for a standard API that any language can implement. Obviously, unions or variadic return types are not a thing in all languages, but the important part is that the returned value can be compared with those defined in here.

In addition, the Parser-API is not made based on the Parser-js. The idea is that Parser-js (and any other parser) will need to adapt it's API to be compliant with this API once we decide it's the right time to do it.

File: docs/v1.md (hunk 3 above)
Comment (smoya) [reply docs/v1.md]: Yeah good catch. Rolling back!

File: docs/v1.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply docs/v1.md:222]: Ok :)

[the remark this answers is not in the document]
Comment (smoya) [reply docs/v1.md:222]: The question now is, should we keep using `undefined` or better to use `null` ? I'm leaning towards the second now (`null`).
cc @jonaslagoni @magicmatatjahu @fmvilas

[the remark this answers is not in the document]
Comment (jonaslagoni) [reply docs/v1.md:222]: To me, `undefined` represent what we want the best, as it means that the property is not defined. With `null` it begins to take a different form and can mean that the value of the property can be `null` as a value 😅

Not that I have a strong opionion, as long as I know what I represent, I dont care either way 😆

[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply docs/v1.md:222]: In the JSON Schema you can also define `null` type -> https://json-schema.org/understanding-json-schema/reference/null.html so `undefined` is better in the case when something isn't defined, but I also have in mind that other languages define only `null` or something similar to indicates that it isn't defined. Hard to say 😅

[the remark this answers is not in the document]
Comment (jonaslagoni) [reply docs/v1.md:222]: As long as it is well documented, it won't matter which one we choose. We explicitly mention for each language what it means there i.e. in some languages `null` will be used (Java) and in others `undefined` (TS).

[the remark this answers is not in the document]
Comment (smoya) [reply docs/v1.md:222]: Fair enough, we stick with `undefined` as it seems to be more clear and does not lead to confusion. Thanks folks :)

**Review by magicmatatjahu — APPROVED** [review]
LGTM!

**Pull request merged**